tunnistamo/tests/test_auth_error_redirect.py

```diff
@@ -0,0 +1,177 @@
+from urllib.parse import urlencode
+
+import pytest
+from django.http import HttpResponse
+from django.test.client import Client
+from django.urls import reverse
+from django.utils.crypto import get_random_string
+from django.utils.translation import gettext_lazy as _
+
+from auth_backends.helsinki_tunnistus_suomifi import HelsinkiTunnistus
+from tunnistamo.tests.conftest import DummyFixedOidcBackend, reload_social_django_utils
+from users.models import LoginMethod
+
+
+def _build_full_url(extra, path):
+    return '{scheme}{host}{port}{path}'.format(
+        scheme=extra.get('wsgi.url_scheme') + '://' if extra.get('wsgi.url_scheme') else '',
+        host=extra.get('SERVER_NAME', ''),
+        port=':' + extra.get('SERVER_PORT') if extra.get('SERVER_PORT') else '',
+        path=path
+    )
+
+
+class CancelSocialAuthBackendRedirectClient(Client):
+    def get(self, path, data=None, follow=False, secure=False, **extra):
+        # Return empty response if the request is to the remote OIDC endpoint
+        if HelsinkiTunnistus().setting('OIDC_ENDPOINT') in _build_full_url(extra, path):
+            return HttpResponse()
+
+        return super().get(path, data=data, follow=follow, secure=secure, **extra)
+
+
+def _start_oidc_authorize(django_client, oidcclient_factory, state=None):
+    LoginMethod.objects.create(
+        provider_id=HelsinkiTunnistus.name,
+        name='Test login method',
+        order=1,
+    )
+
+    redirect_uris = ['https://example.com/callback']
+    oidc_client = oidcclient_factory(redirect_uris=redirect_uris)
+
+    # This will redirect the user to the Tunnistamo login view and from there to the
+    # "Test login method". We cancel the last redirect in the
+    # CancelSocialAuthBackendRedirectClient. The cancelled redirect is required to
+    # have the "next" parameter in the django_clients session.
```

Comment:
While this comment is helpful, I think it's in a wrong location.

The first sentence (This will redirect the user to the Tunnistamo login view and from there to the
"Test login method".) feels like it should be this method's comment, so right below the `def _start_oidc_authorize...`.

The rest feels like it should be within the `CancelSocialAuthBackendRedirectClient` class, perhaps replacing the current comment that is there.